From 95ed7b0ccf74cdc2423d3a44d776be09544ed3f9 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 02 Feb 2023 09:20:07 +0200
Subject: [PATCH] [SW-0] gaudi2: initialize arc0 dccm only in cold reset.

The CCM memories are erased only in cold reset hence no
need to initialize this in other resets. This patch adds
support to clear DCCM memory only in cold reset.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I86b23323267b2c06b24c2afa87d63abe2edb32f7
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index cd2247c..6c715ae 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -169,14 +169,14 @@
 	regmap_writel(ARC_RST_VEC_ADDR, mmPSOC_ARC0_CFG_RST_VEC_ADDR);
 	regmap_writel(ARC_RST_VALUE, mmPSOC_ARC0_CFG_ARC_RST);
 
-	/* Clear DCCM memory */
-	init_arc_mem(0, ARC0_DCCM_FULL_BASE, ARC0_DCCM_SIZE);
-
-	/* Clear ARC0 ICCM memory on cold reset only */
+	/* Clear ARC0 ICCM and DCCM memory on cold reset only */
 	rst_src = hl_rst_src_get();
 	if (rst_src != HL_COLD_RST)
 		return;
 
+	/* Clear DCCM memory */
+	init_arc_mem(0, ARC0_DCCM_FULL_BASE, ARC0_DCCM_SIZE);
+	/* Clear ICCM memory */
 	init_arc_mem(0, ARC0_ICCM_FULL_BASE, ARC0_ICCM_SIZE);
 }
 
